db/hints/internal/hint_host_manager.cc

```diff
@@ -834,5 +834,9 @@ resource_manager& end_point_hints_manager::shard_resource_manager() {
     return _shard_manager._resource_manager;
 }
 
+bool end_point_hints_manager::sender::replay_allowed() const noexcept {
```

Comment:
I agree the commit title could be better. The premise of these changes is the fact that we cannot have a cycle in `#include`s. Because the declaration of `sender` is moved to a separate header file in the following commit, and `hint_host_manager.hh` includes that header, `sender` has to rely on `end_point_hints_manager` being an incomplete type in its declaration. A consequence of that is that we won't be able to access methods of `end_point_hints_manager` from `hint_sender.hh`. These changes prepare the ground for it, but maybe doing so while moving the declaration of `sender` would've been a better idea because the reason why we do that would be clearer.

Do you think I should reword the commit message or squash the commits?

Reply:
Please reword the commit message. The title could just be that you are moving the implementation from header to cc file, and the rest of the message should explain the rationale behind it (you are going to move the declaration of `end_point_hints_manager` to a separate header and don't want to include it; `sender::replay_allowed` needs endpoint hint manager's definition so you need to move it to the cc file, etc.).